### DISCUSSION TITLE
How to get DEPHY-SCM directory

### DISCUSSION THREAD

**Discussion by bluefinweiwei** [NCAR/ccpp-scm#301] [Cases, Initial Conditions, Forcing]
Hi @grantfirl, I am trying to test physics with the DYNAMO case(s). Under case_config directory, I see there are two files seemingly relevant, i.e., DYNAMO_NSA3a.nml and DYNAMO_NSA3a_D1.nml. In these two namelist files, I was pointed to find supported data (presumably IC and forcing) under ../data/DEPHY-SCM/DYNAMO/NSA3a and ../data/DEPHY-SCM/DYNAMO/NSA3a_D1. But no DEPHY-SCM direcotry could be found under data, and I don't see anything relevant under data/processed_case_input either. Would you please provide some guidance? Thanks a lot!

**Comment by grantfirl** [comment]
@bluefinweiwei Sure, please clone the DEPHY-SCM repo under the data directory:

`cd into scm/data`
`git clone -b master https://github.com/GdR-DEPHY/DEPHY-SCM`

The top of master of this repo has updated to the latest DEPHY-SCM v1 which is much newer than what the current SCM code supports, so you need to checkout a specific commit of DEPHY-SCM to get cases that should work:

`git checkout f65416`

This should correspond to Jan. 13, 2021, roughly when the SCM was coded to work with this repo. Please let me know if you run into any errors.

**Reply by bluefinweiwei** [reply]
Thanks for the quick guidance @grantfirl! I have been using a pretty old version of ccpp-scm (HEAD e0a0bc; last commit by you on Sept 24 of last year) to work with the prototype physics, which is absolutely not new but has less hassle. I'll pull DEPHY-SCM and let you of any problems.

**Reply by bluefinweiwei** [reply]
Hello @grantfirl When I execute ./run_scm.py -c DYNAMO_NSA3a -s SCM_GFS_v16 --runtime 86400, I see the following error which I have no clue how to proceed (related to surface fluxes). I tried to add the following lines in the namelist:
sfc_flux_spec = .true.,
sfc_roughness_length_cm = 0.02,
but problem remained...
**FYI: I'm using the top _(main)_ branch of ccpp-scm as of today.** 
**Could you please help? Thanks a lot!**
INFO: Setting up experiment DYNAMO_NSA3a with suite SCM_GFS_v16 using the default namelist and tracers for the suite
INFO: Namelist runtime adjustment 86400 IS applied
INFO: Parsing case configuration namelist ../etc/case_config/DYNAMO_NSA3a.nml
Traceback (most recent call last):
  File "./run_scm.py", line 496, in <module>
    main()
  File "./run_scm.py", line 482, in main
    exp_dir = exp.setup_rundir()
  File "./run_scm.py", line 291, in setup_rundir
    if surface_flux_spec:
**UnboundLocalError: local variable 'surface_flux_spec' referenced before assignment**
